### PR TITLE
STITCH-1371 Implement HTTP service in Core and iOS

### PR DESCRIPTION
**Pull request by adamchel**
almost no drivebys this time. `run_ios_tests` is still broken until we get them to reboot the box, but everything passes for me locally.

### CODE REVIEW

File: StitchCoreServicesHttp-iOS/StitchCoreServicesHttp-iOS/StitchCoreServicesHttp_iOS.h [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#import <UIKit/UIKit.h>
+
+//! Project version number for StitchCoreServicesHttp_iOS.
+FOUNDATION_EXPORT double StitchCoreServicesHttp_iOSVersionNumber;
+
+//! Project version string for StitchCoreServicesHttp_iOS.
+FOUNDATION_EXPORT const unsigned char StitchCoreServicesHttp_iOSVersionString[];
+
+// In this header, you should import all the public headers of your framework using statements like #import <StitchCoreServicesHttp_iOS/PublicHeader.h>
```
Comment (jsflax) [inline StitchCoreServicesHttp-iOS/StitchCoreServicesHttp-iOS/StitchCoreServicesHttp_iOS.h]: [nit] Can remove these newlines. We can probably start removing these headers as well.

File: StitchCoreServicesHttp/Sources/StitchCoreServicesHttp/HttpResponse.swift [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+                }
+                
+                guard let value: String = try? document.get("value") else {
+                    throw MongoError.typeError(message: "expected string value for cookie value in HTTP service response")
+                }
+                
+                let keys = document.keys
+                
+                var path: String?
+                if keys.contains(HttpCookie.CodingKeys.path.rawValue) {
```
Comment (jsflax) [inline StitchCoreServicesHttp/Sources/StitchCoreServicesHttp/HttpResponse.swift]: [minor] Unless `Documents` operate radically differently than dictionaries (which they shouldn't given the try notation), you can remove all of these conditionals. Each one could be like:
`let path = try? document[HttpCookie.CodingKeys.path.rawValue]`.

Comment (adamchel) [reply StitchCoreServicesHttp/Sources/StitchCoreServicesHttp/HttpResponse.swift]: The syntax I was using is for the sake of type safety, but I played around with it more and made it cleaner like this:

```swift
let path = document[HttpCookie.CodingKeys.path.rawValue] as? String
```

File: StitchCoreServicesHttp-iOS/StitchCoreServicesHttp-iOS/StitchCoreServicesHttp_iOS.h (hunk 1 above)
Comment (adamchel) [reply StitchCoreServicesHttp-iOS/StitchCoreServicesHttp-iOS/StitchCoreServicesHttp_iOS.h]: Removed the newlines. Are these headers for the sake of Objective-C compatibility? Happy to remove them if we are definitely not supporting objc

**Pull request merged**